endpoints/openrtb2/auction.go

```diff
@@ -335,28 +337,49 @@ func sendAuctionResponse(
 //
 // If the errors list has at least one element, then no guarantees are made about the returned request.
 func (deps *endpointDeps) parseRequest(httpRequest *http.Request, labels *metrics.Labels, hookExecutor hookexecution.HookStageExecutor) (req *openrtb_ext.RequestWrapper, impExtInfoMap map[string]exchange.ImpExtInfo, storedAuctionResponses stored_responses.ImpsWithBidResponses, storedBidResponses stored_responses.ImpBidderStoredResp, bidderImpReplaceImpId stored_responses.BidderImpReplaceImpID, account *config.Account, errs []error) {
-	req = &openrtb_ext.RequestWrapper{}
-	req.BidRequest = &openrtb2.BidRequest{}
 	errs = nil
-
-	// Pull the request body into a buffer, so we have it for later usage.
-	lr := &io.LimitedReader{
-		R: httpRequest.Body,
-		N: deps.cfg.MaxRequestSize,
+	var limitedReqReader *io.LimitedReader
+	reqContentEnc := httpRequest.Header.Get("Content-Encoding")
+	if reqContentEnc == "gzip" {
+		if !deps.cfg.Compression.Request.Enabled ||
+			deps.cfg.Compression.Request.CType != config.CompressionTypeGZIP {
+			errs = []error{fmt.Errorf("Content-Encoding of type %s is not supported", reqContentEnc)}
+			return
+		}
+		if gzReader, err := gzip.NewReader(httpRequest.Body); err == nil {
+			limitedReqReader = &io.LimitedReader{
+				R: gzReader,
+				N: deps.cfg.MaxRequestSize,
+			}
+			defer gzReader.Close()
+		} else {
+			errs = []error{err}
+			return
+		}
+	} else {
+		limitedReqReader = &io.LimitedReader{
+			R: httpRequest.Body,
+			N: deps.cfg.MaxRequestSize,
+		}
 	}
```

Comment:
Please extract out the limited reader initialization, so we don't need to repeat that code. Consider a helper method that inspects the request and returns a reader, either `gzReader` or  `httpRequest.Body`.